### PR TITLE
DR-125: add to and from table ids to relationship model

### PR DESCRIPTION
**Pull request by ahaessly**

### CODE REVIEW

**Review by thathert — CHANGES_REQUESTED** [review]
One other request from the ticket: Can you store a link to the StudyTable in the StudyRelationship metadata class?

File: src/main/java/bio/terra/dao/AssetDao.java [hunk 1]
```diff
@@ -102,15 +102,15 @@ public void retrieve(Study study) {
         MapSqlParameterSource params = new MapSqlParameterSource().addValue("studyId", study.getId());
 
         return jdbcTemplate.query(sql, params, (rs, rowNum) -> {
-            UUID specId = UUID.fromString(rs.getString("id"));
+            UUID specId = rs.getObject("id", UUID.class);
```
Comment (thathert) [inline src/main/java/bio/terra/dao/AssetDao.java:105]: Awesome, this is a big improvement.

**Review by thathert — APPROVED** [review]
LGTM

File: src/main/java/bio/terra/dao/AssetDao.java [hunk 2]
```diff
@@ -126,31 +126,30 @@ public void retrieve(Study study) {
...
-                    assetTable.getColumns().add(newColumn);
-                });
+        String sql = "SELECT asset_column.id, asset_column.study_column_id, study_column.table_id " +
+                "FROM asset_column " +
+                "INNER JOIN study_column ON asset_column.study_column_id = study_column.id " +
+                "WHERE asset_id = :assetId";
+        MapSqlParameterSource params = new MapSqlParameterSource().addValue("assetId", spec.getId());
+        List<Map<String, Object>> results = jdbcTemplate.queryForList(sql, params);
+        results.forEach(rs -> {
+            UUID tableId = (UUID) rs.get("table_id");
```
Comment (ddietterich) [inline src/main/java/bio/terra/dao/AssetDao.java:136]: Should these be getObject(... UUID.class) too?
I'm not sure what the best practice is.

**Review by ddietterich — APPROVED** [review]
One comment - I think you might have missed a case of UUID conversion, but not sure.
Otherwise, LGTM

File: src/main/java/bio/terra/dao/AssetDao.java (hunk 2 above)
Comment (ahaessly) [reply src/main/java/bio/terra/dao/AssetDao.java:136]: the variable name rs is misleading. i'm looping through a regular Map<String,Object> and map does not have a getObject method.

**Pull request merged**